src/main/java/org/openrewrite/java/spring/boot2/WebSecurityConfigurerAdapterRecipe.java

```diff
@@ -0,0 +1,240 @@
+                            b = b.withStatements(ListUtils.concat(b.getStatements(), autoFormat(returnStatement, context)));
+//                        JavaTemplate template = JavaTemplate.builder(() -> getCursor(), "return " + ((J.VariableDeclarations)parentMethod.getParameters().get(0)).getVariables().get(0).getSimpleName() + ".build();").javaParser(() -> JavaParser.fromJavaVersion().build()).build();
+//                        b = b.withTemplate(template, b.getCoordinates().addStatement(null));
+                        } else if (CONFIGURE_WEB_SECURITY_METHOD_MATCHER.matches(parentMethod, classDecl)) {
+                            J.Lambda.Parameters parameters = new J.Lambda.Parameters(Tree.randomId(), Space.EMPTY, Markers.EMPTY, true, parentMethod.getParameters().stream().filter(J.VariableDeclarations.class::isInstance).map(J.VariableDeclarations.class::cast).map(p -> {
+                                J.VariableDeclarations.NamedVariable varDecl = p.getVariables().get(0);
+                                return new JRightPadded<J>(varDecl.getName(), Space.EMPTY, Markers.EMPTY);
+                            }).collect(Collectors.toList()));
+
+                            J.Lambda lambda = new J.Lambda(Tree.randomId(), Space.build(" ", Collections.emptyList()), Markers.EMPTY, parameters, Space.build(" ", Collections.emptyList()), b.getStatements().size() == 1 ? b.getStatements().get(0).withPrefix(Space.EMPTY) : b, null);
+                            J.Return returnStatement = new J.Return(Tree.randomId(), Space.build("\n", Collections.emptyList()), Markers.EMPTY, lambda);
+                            b = b.withStatements(Collections.singletonList(autoFormat(returnStatement, context)));
+                        }
+                    }
+                }
+                return b;
+            }
+
+            private J.MethodDeclaration addBeanAnnotation(J.MethodDeclaration m, Cursor c) {
+                maybeAddImport(FQN_BEAN);
+                JavaTemplate template = JavaTemplate.builder(() -> c, BEAN_ANNOTATION).imports(FQN_BEAN).javaParser(() -> JavaParser.fromJavaVersion()
+                        .dependsOn("package " + BEAN_PKG + "; public @interface " + BEAN_SIMPLE_NAME + " {}").build()).build();
+                return m.withTemplate(template, m.getCoordinates().addAnnotation(null));
+            }
+
+            private boolean isApplicableClass(J.ClassDeclaration classDecl) {
+                if (classDecl.getExtends() != null) {
+                    JavaType.FullyQualified superType = TypeUtils.asFullyQualified(classDecl.getExtends().getType());
+                    if (superType != null && FQN_WEB_SECURITY_CONFIGURER_ADAPTER.equals(superType.getFullyQualifiedName())
+                            && classDecl.getLeadingAnnotations().stream().anyMatch(a -> TypeUtils.isOfClassType(a.getType(), FQN_CONFIGURATION))) {
```

Comment:
Consider simplifying this with `TypeUtils#isAssignableTo`
```suggestion
                if (TypeUtils.isAssignableTo(FQN_WEB_SECURITY_CONFIGURER_ADAPTER, classDecl.getType())
                    && classDecl.getLeadingAnnotations().stream().anyMatch(a -> TypeUtils.isOfClassType(a.getType(), FQN_CONFIGURATION))) {
```